### DISCUSSION TITLE
`ChatCompletionClient` cannot be used in a standard compliant system due to exception access modifiers + inconsistent service exception access modifiers.

### DISCUSSION THREAD

**Discussion by Gummimundur** [hassanhabib/Standard.AI.OpenAI#235] [General]
@hassanhabib @ElbekDeveloper @BrianLParker 

Some of the `ChatCompletionClient` exceptions are `internal` and others are `public` therefore the following problem occurs when trying to integrate this library in a standard compliant system. The `ChatCompletionClient` exceptions should all be public, right?

![image](https://user-images.githubusercontent.com/22801304/228498070-d7a8e48a-6d50-4598-964e-6bed2f5e97ab.png)

Also there are inconsistent access modifiers in the service exceptions. For example `InvalidConfigurationChatCompletionException` is internal while `InvalidConfigurationCompletionException` is public. Which one should it be? Should all the service level exceptions be `public`? All `internal`? Or should it depend on something else?

**Comment by hassanhabib** [comment]
Yes all exceptions should be public.
